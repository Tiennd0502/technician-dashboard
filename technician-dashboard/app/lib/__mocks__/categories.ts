export const CATEGORIES = [
  {
    id: '01',
    label: 'Sofa Cleaning',
    image: 'https://i.ibb.co/QcSmTHW/category01.webp',
  },
  {
    id: '02',
    label: 'Bathroom',
    image: 'https://i.ibb.co/KDQzZ3s/category02.web',
  },
  {
    id: '03',
    label: 'Best Control',
    image: 'https://i.ibb.co/jwsPTTt/category03.webp',
  },
  {
    id: '04',
    label: 'View All',
    image: 'https://i.ibb.co/KxsM943/category04.webp',
  },
];
