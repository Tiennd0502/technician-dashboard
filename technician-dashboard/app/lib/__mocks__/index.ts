export * from './table';
export * from './categories';
export * from './services';
