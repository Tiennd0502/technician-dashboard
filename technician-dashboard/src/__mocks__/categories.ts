export const CATEGORIES = [
  {
    id: '01',
    label: 'Sofa Cleaning',
    image: 'https://i.ibb.co/dKnNFV1/category01.png',
  },
  {
    id: '02',
    label: 'Bathroom',
    image: 'https://i.ibb.co/Qdtfx8N/category02.png',
  },
  {
    id: '03',
    label: 'Best Control',
    image: 'https://i.ibb.co/dDLCwNf/category03.png',
  },
  {
    id: '04',
    label: 'View All',
    image: 'https://i.ibb.co/cDz3H5Q/category04.png',
  },
];
