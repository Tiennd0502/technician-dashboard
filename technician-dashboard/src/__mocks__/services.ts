export const SERVICE = {
  id: '01',
  name: 'Carpenter',
  price: 45,
  image: 'https://i.ibb.co/7WfPDMJ/service01.png',
};

export const SERVICES = [
  {
    id: '01',
    name: 'Carpenter',
    price: 45,
    image: 'https://i.ibb.co/7WfPDMJ/service01.png',
  },
  {
    id: '02',
    name: 'Ac Service & repair',
    price: 25,
    image: 'https://i.ibb.co/G069RKZ/service02.png',
  },
  {
    id: '03',
    name: 'Kitchecn Deep Cleaning',
    price: 30,
    image: 'https://i.ibb.co/nw3YSHS/service03.png',
  },
  {
    id: '04',
    name: 'Electrician',
    price: 50,
    image: 'https://i.ibb.co/Cvnw8GY/service04.png',
  },
];
