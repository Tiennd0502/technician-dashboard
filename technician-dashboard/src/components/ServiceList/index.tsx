import { memo } from 'react';
import { Flex } from '@chakra-ui/react';

// Constants
import { CURRENCY_CHARACTERS } from '@/constants';

// Types
import { Service } from '@/interfaces';

// Components
import { ServiceCard } from '..';

interface ServiceListProps {
  list: Service[];
}

const ServiceList = ({ list }: ServiceListProps) => (
  <Flex gap='5' role='list'>
    {list.map((item) => (
      <ServiceCard key={item.id} service={item} currencyCharacter={CURRENCY_CHARACTERS.EURO} />
    ))}
  </Flex>
);

export default memo(ServiceList);
