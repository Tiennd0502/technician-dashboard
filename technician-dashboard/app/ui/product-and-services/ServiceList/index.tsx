import { memo } from 'react';
import { Flex } from '@chakra-ui/react';

import { CURRENCY_CHARACTERS } from '@/lib/constants';
import { Service } from '@/lib/interfaces';
import { ServiceCard } from '..';

interface ServiceListProps {
  list: Service[];
}

const ServiceList = ({ list }: ServiceListProps) => (
  <Flex gap='5' data-testid='list-services'>
    {list.map((item) => (
      <ServiceCard key={item.id} service={item} currencyCharacter={CURRENCY_CHARACTERS.EURO} />
    ))}
  </Flex>
);

export default memo(ServiceList);
