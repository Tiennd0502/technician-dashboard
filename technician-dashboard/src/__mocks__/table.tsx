import { STATUS } from '@/interfaces';
import { StatusLabel } from '@/components';

export const PRODUCTS = [
  {
    id: '1',
    name: 'Washing machine',
    brand: 'Hitachi',
    service: 'Repair',
    status: 1,
  },
  {
    id: '2',
    name: 'Air conditioner',
    brand: 'LG',
    service: 'Installation',
    status: 0,
  },
  {
    id: '3',
    name: 'Smart TV',
    brand: 'OnePlus',
    service: 'Uninstallation',
    status: 1,
  },
  {
    id: '4',
    name: 'Water purifier',
    brand: 'Kent',
    service: 'Repair',
    status: 1,
  },
];

const customViewStatus = (value: string | number | boolean) => (
  <StatusLabel value={value as STATUS} />
);

export const PRODUCT_HEADER_COLUMNS = [
  {
    key: 'checkbox',
    label: 'Checkbox',
    width: {
      md: '30px',
      lg: '50px',
    },
    isCheckbox: true,
  },
  {
    key: 'name',
    label: 'Product name',
    width: {
      md: '100%',
      lg: '28%',
    },
    onSort: () => null,
  },
  {
    key: 'brand',
    label: 'Brand name',
    width: {
      md: '100%',
      lg: '17%',
    },
  },
  {
    key: 'service',
    label: 'Service',
    width: {
      md: '100%',
      lg: '15%',
    },
  },
  {
    key: 'status',
    label: 'Status',
    width: {
      md: '100%',
      lg: '18%',
    },
    customView: customViewStatus,
  },
  {
    key: 'actions',
    label: 'Actions',
    width: {
      md: '100%',
      lg: 'calc(32% - 35px)',
    },
    isAction: true,
  },
];
