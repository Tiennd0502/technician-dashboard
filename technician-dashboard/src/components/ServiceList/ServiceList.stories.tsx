import type { Meta, StoryObj } from '@storybook/react';

import { SERVICES } from '@/__mocks__';

import ServiceList from '.';

const meta: Meta<typeof ServiceList> = {
  title: 'Components/ServiceList',
  component: ServiceList,
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof ServiceList>;

export const Default: Story = {
  args: { list: SERVICES },
};
