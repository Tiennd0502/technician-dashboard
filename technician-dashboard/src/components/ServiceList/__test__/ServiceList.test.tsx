import { render } from '@testing-library/react';

import { SERVICES } from '@/__mocks__';
import ServiceList from '..';

describe('ServiceList test cases', () => {
  const props = {
    list: SERVICES,
  };

  test('should render correctly', () => {
    const { container } = render(<ServiceList {...props} />);
    expect(container).toMatchSnapshot();
  });

  test('Should render enough elements', () => {
    const { getByRole } = render(<ServiceList {...props} />);
    const serviceList = getByRole('list');

    expect(serviceList.childNodes).toHaveLength(SERVICES.length);
  });

  test('Should render empty elements', () => {
    const { getByRole } = render(<ServiceList list={[]} />);
    const serviceList = getByRole('list');

    expect(serviceList.childNodes).toHaveLength(0);
  });
});
