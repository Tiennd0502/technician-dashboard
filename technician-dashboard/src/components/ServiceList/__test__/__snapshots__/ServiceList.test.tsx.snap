// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceList test cases should render correctly 1`] = `
<div>
  <div
    class="css-1j52cd2"
    role="list"
  >
    <a
      class="chakra-link css-1r22un5"
    >
      <img
        alt="Carpenter"
        class="chakra-image css-1fayruo"
        src="https://i.ibb.co/7WfPDMJ/service01.png"
      />
      <div
        class="css-1g4yje1"
        justify-content="flex-end"
      >
        <div
          class="css-17xejub"
        />
        <p
          class="chakra-text css-od0eia"
        >
          Carpenter
        </p>
        <p
          class="chakra-text css-18bew60"
        >
          €45
        </p>
      </div>
    </a>
    <a
      class="chakra-link css-1r22un5"
    >
      <img
        alt="Ac Service & repair"
        class="chakra-image css-1fayruo"
        src="https://i.ibb.co/G069RKZ/service02.png"
      />
      <div
        class="css-1g4yje1"
        justify-content="flex-end"
      >
        <div
          class="css-17xejub"
        />
        <p
          class="chakra-text css-od0eia"
        >
          Ac Service & repair
        </p>
        <p
          class="chakra-text css-18bew60"
        >
          €25
        </p>
      </div>
    </a>
    <a
      class="chakra-link css-1r22un5"
    >
      <img
        alt="Kitchecn Deep Cleaning"
        class="chakra-image css-1fayruo"
        src="https://i.ibb.co/nw3YSHS/service03.png"
      />
      <div
        class="css-1g4yje1"
        justify-content="flex-end"
      >
        <div
          class="css-17xejub"
        />
        <p
          class="chakra-text css-od0eia"
        >
          Kitchecn Deep Cleaning
        </p>
        <p
          class="chakra-text css-18bew60"
        >
          €30
        </p>
      </div>
    </a>
    <a
      class="chakra-link css-1r22un5"
    >
      <img
        alt="Electrician"
        class="chakra-image css-1fayruo"
        src="https://i.ibb.co/Cvnw8GY/service04.png"
      />
      <div
        class="css-1g4yje1"
        justify-content="flex-end"
      >
        <div
          class="css-17xejub"
        />
        <p
          class="chakra-text css-od0eia"
        >
          Electrician
        </p>
        <p
          class="chakra-text css-18bew60"
        >
          €50
        </p>
      </div>
    </a>
  </div>
</div>
`;
